jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/editor/GettingStartedPanelUtils.kt

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.gettingstarted.editor
+
+import com.intellij.openapi.project.Project
+import com.intellij.ui.dsl.builder.Panel
+import software.aws.toolkits.core.credentials.CredentialIdentifier
+import software.aws.toolkits.jetbrains.core.credentials.AwsBearerTokenConnection
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionState
+import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitAuthManager
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.lazyIsUnauthedBearerConnection
+import software.aws.toolkits.jetbrains.core.credentials.pinning.CodeCatalystConnection
+import software.aws.toolkits.jetbrains.core.credentials.pinning.CodeWhispererConnection
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_URL
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+
+
+data class ValidActiveIamConnection(
+    val isConnectionValid: ValidConn,
+    val activeConnection: CredentialIdentifier?,
+    val connectionType: CWConnectionType?
+)
+
+data class ValidActiveConnection (
+    val isConnectionValid: ValidConn,
+    val activeConnection: AwsBearerTokenConnection?,
+    val connectionType: CWConnectionType?
+)
```

Comment:
can probably model this a bit better

maybe like:
```kt
sealed interface ActiveConnection {
    object Expired : ActiveConnection

    object NotConnected : ActiveConnection

    data class ValidBearer(
        val activeConnection
        ...
    ) : ActiveConnection

    data class ValidIam(
        val activeConnection
        ...
    ) : ActiveConnection
}
```

and then you don't need the enum anymore